src/domain/datasets/domain/src/entities/dataset_entry.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use opendatafabric::{AccountID, DatasetID, DatasetName};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct DatasetEntry {
+    pub id: DatasetID,
+    pub owner_id: AccountID,
+    pub name: DatasetName,
```

Comment:
Why do we need to store the name here? It does not look like it's used in lookups.

Reply:
> Why do we need to store the name here?

Yes, we need it

> It does not look like it's used in lookups.

Added `DatasetEntryRepository::get_dataset_entry()` method -- Fixed